AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -235,7 +235,538 @@ def __repr__(self):
     return '<PushQueue {}: {}>'.format(self.name, attr_str)
 
 
+class PostgresPullQueue(Queue):
+
+  """
+  Before using Postgres implementation, make sure that
+  connection using appscale user can be created:
+  /etc/postgresql/9.5/main/pg_hba.conf
+  """
+
+  def __init__(self, queue_info, app, pg_connection=None):
+    """ Create a PostgresPullQueue object.
+
+    Args:
+      queue_info: A dictionary containing queue info.
+      app: A string containing the application ID.
+      pg_connection: A psycopg2 connection to PostgreSQL.
+    """
+    super(PostgresPullQueue, self).__init__(queue_info, app)
+    self.pg_connection = pg_connection
+    self.pg_cursor = pg_connection.cursor()
+    self.ensure_tables_created()
+
+  @property
+  def tasks_table_name(self):
+    return 'pullqueue-{}'.format(self.name)
+
+  def ensure_tables_created(self):
+    try:
+      self.pg_cursor.execute(
+        'CREATE TABLE IF NOT EXISTS "{table_name}" ('
+        '  task_name varchar(500) NOT NULL,'
+        '  time_enqueued timestamp NOT NULL,'
+        '  retry_count integer NOT NULL,'
```

Comment:
It might make sense to name this lease_count since it's a little more explicit about what it's indicating and it's a little easier to reason about. For example, if the task has been leased exactly once, does that mean that the task has been (re)tried?